pkg/generate/app/imagestreamlookup.go

```diff
@@ -162,10 +162,14 @@ func (r ImageStreamSearcher) Search(precise bool, terms ...string) (ComponentMat
 							continue
 						}
 
-						// If the user specified a tag in their search string (followTag == false),
-						// then score this match lower.
-						tagScore := score
+						// at best this is a partial match situation.  The user didn't
+						// specify a tag, so we tried "latest" but could not find an image associated
+						// with the latest tag (or one that is followed by the latest tag), or
+						// they specified a tag that we could not find.
+						tagScore := score + 0.1
 						if !followTag {
+							// If the user specified a tag in their search string (followTag == false),
+							// then score this match lower.
 							tagScore += 0.5
```

Comment:
@bparees I agree with always setting tagScore := score + 0.5 regardless of the value of followTag.  I think the intended behaviour it changes is when someone does `oc new-app builder~repo`, where no tag is specified as part of `builder` and no "latest" image is known to us.  In that case, currently we'll go with any other tag we find.  I think your proposal would mean we'd print the options and stop, which doesn't seem like a bad plan.